apps/judicial-system/backend/src/app/modules/verdict/verdict.service.ts

```diff
@@ -139,10 +139,19 @@ export class VerdictService {
     return this.sequelize.transaction(async (transaction) => {
       return await Promise.all(
         verdicts.map((verdict) => {
-          // Only the latest verdict is relevant
-          const currentVerdict = defendants?.find(
+          const currentDefendant = defendants?.find(
             (defendant) => verdict.defendantId === defendant.id,
-          )?.verdicts?.[0]
+          )
+
+          // This prevents the creation of verdicts for defendants that do not belong to the case
+          if (!currentDefendant) {
```

Comment:
There was a flaw in the logic here as it was possible to create a verdict for a defendant in another case. Not critical, but now impossible.